### ISSUE TITLE
Android: "UTM Visited" event not logged when clicking link while app is backgrounded

### ISSUE THREAD

**Issue by marcshilling** [CleverTap/clevertap-react-native#400]
I am trying to set up UTM tracking for my app so that we can measure clicks across iOS/Android (universal links) and web.

To test this, I can just refresh my user on the CleverTap dashboard and see the event come in almost instantly:
![image](https://github.com/CleverTap/clevertap-react-native/assets/5598961/a93d9c46-08f4-4d6e-9192-f865705331e2)

This seems to be working well in all cases, except when the Android app is already "alive" in the background. The "UTM Visted" event is sent properly when the app is quit and does a cold boot, but if I background the app and click the link it is not sent. On iOS, the event is sent properly in both cases.

More info:
* `clevertap-react-native v2.2.1`
* I've added `implementation 'com.android.installreferrer:installreferrer:2.2'` to my build.gradle per the [docs](https://developer.clevertap.com/docs/android-advanced-features#referral-tracking) (this was necessary to get the cold-boot tracking to work)

**Comment by marcshilling** [comment]
> [@marcshilling](https://github.com/marcshilling) Apologies for the delayed response? Are you still facing this issue?

Yes, it's still an issue. In fact, when I went to test this again, I actually can't get the "UTM Visited" event to appear at all now, either with a cold boot or opening from the background.

**Comment by Anush-Shand** [comment]
@marcshilling 
Can you explain the use-case
Also please confirm if you have followed these docs https://developer.clevertap.com/docs/android-advanced-features#referral-tracking

You can test the same using the hack as mentioned here https://stackoverflow.com/questions/49040093/testing-google-play-install-referrer-library

**Comment by Anush-Shand** [comment]
Closing this due to inactivity. Feel free to reopen
